Инкремент 15

Fairness2 opened the pull request:
Добавлено использование пула подключения к серверу, новые метрики использования системы. Проверка подписи запроса

Fairness2 committed:
iter15 | Добавил хэширование тела запроса и обновил обработку тела метрик.

Добавил хэширование тела запроса с использованием HMAC-SHA256 и поддержку ключа шифрования. Также реорганизован код для подготовки тела метрик, включая сжатие и маршалинг.

Список измененных файлов:
- cmd/agent/collector/sender/sender.go
- cmd/agent/config/env.go
- cmd/agent/config/parse.go

Fairness2 committed:
iter14 | Добавил хэширование тела запроса и обновил обработку тела метрик.

Добавил хэширование тела запроса с использованием HMAC-SHA256 и поддержку ключа шифрования. Также реорганизован код для подготовки тела метрик, включая сжатие и маршалинг.

Список измененных файлов:
- cmd/agent/collector/sender/sender.go
- cmd/agent/config/env.go
- cmd/agent/config/parse.go

Fairness2 committed:
iter14.1 | Добавлена проверка подписи тела запроса и обработка хеширования

Добалено новое middleware для проверки подписи тела запроса на сервере. Изменена логика хеширования тела в агента и добавлены тесты для новой middleware.

Список изменённых файлов:
- cmd/agent/collector/sender/sender.go
- internal/middlewares/middlewares.go
- cmd/server/config/config.go
- cmd/agent/config/parse.go
- cmd/server/config/parse.go
- internal/middlewares/middlewares_test.go
- cmd/server/main.go

Fairness2 committed:
Merge remote-tracking branch 'origin/iter14' into iter14

# Conflicts:
#	cmd/agent/collector/sender/sender.go
#	cmd/agent/config/parse.go

Fairness2 committed:
iter14.2 | Оптимизирована обработка тела запроса и добавлены комментарии к тестам.

Переработана функция упаковки и сжатия тела запроса в агенте. Добавлены комментарии к тестам и перераспределение миддлварей на сервере.

Список изменённых файлов:
- cmd/agent/collector/sender/sender.go
- internal/middlewares/middlewares_test.go
- cmd/server/main.go

Fairness2 committed:
iter14.3 | Изменена логика проверки хэш-заголовка и обновлены тесты.

При отсутствии заголовка с хэшем, не проверяем тело на подпись

Список изменённых файлов:
- internal/middlewares/middlewares.go
- internal/middlewares/middlewares_test.go

Fairness2 committed:
Merge branch 'refs/heads/iter13' into iter15

# Conflicts:
#	cmd/agent/collector/sender/sender.go
#	cmd/agent/config/env.go
#	cmd/agent/config/parse.go

Fairness2 committed:
iter15 | Реализован асинхронный механизм сохранения сообщений.

Добавлен канал для обработки сообщений и метод групповой вставки сообщений в базу данных.

Список изменённых файлов:
- internal/middlewares/middlewares.go
- internal/store/skill/pg/store.go
- internal/store/skill/mock/store.go
- cmd/skill/app.go
- internal/store/skill/store.go

Fairness2 committed:
iter15  | Убраны методы сжатия и хеширования из отправщика и перемещены в новый пул отправщиков

Вместо прямого отправления метрик с их сжатием и хешированием в клиенте отправки, создан новый пул отправщиков, который берет на себя эти задачи. Это улучшает масштабируемость и распределение нагрузки при отправке данных на сервер, так как теперь используется пул горутин.

Список измененных файлов:
- cmd/agent/collector/sender/sender.go
- cmd/agent/collector/collection/collection.go
- cmd/agent/sendpool/pool.go
- cmd/agent/config/env.go
- cmd/agent/config/parse.go
- cmd/agent/main.go

Fairness2 committed:
iter15.2 | Добавлен сбор системных метрик и рефакторинг клиента-отправителя.

Добавлен сбор метрик использования системы и интерфейс для пула подключений в модуле sender. Интерфейс Pusher заменил конкретную реализацию sendpool.Pool. Очистка зависимости и улучшение метрик.

Измененные файлы:
- cmd/agent/collector/sender/sender.go
- cmd/agent/collector/collection/initializer.go
- go.sum
- cmd/agent/collector/collection/collection.go
- go.mod
- cmd/agent/collector/sender/mock/pusher.go

Fairness2 committed:
iter15.3 | Обновлены версии контейнеров для GitHub Actions.

Обновлены версии контейнеров Golang с 1.22 до 1.23 в workflows metricstest и statictest. Это позволит использовать последние исправления и улучшения языка.

Измененные файлы:
- .github/workflows/mertricstest.yml
- .github/workflows/statictest.yml

Fairness2 commented:
СтатикТест упал с паникой `panic: *types.Alias: any`, странно. Как починят, так перепрогоню и поправлю, что обнаружит

Fairness2 committed:
Revert "iter15.3 | Обновлены версии контейнеров для GitHub Actions."

This reverts commit b8ffc1bdc80fef5b88ce037680ec597fbbf8eb9f.

vokinneberg commented on cmd/agent/collector/collection/initializer.go:
```diff
@@ -49,3 +53,43 @@ func collect() {
+	if err != nil {
+		logger.Log.Error(err)
+	}
+	for i, percent := range cpuStat {
```
👍

vokinneberg commented on cmd/agent/collector/sender/sender.go:
```diff
@@ -15,30 +12,26 @@ import (
 
-func New(mCollection *collection.Type) *Client {
+// Pusher интерфейс для пула конектов к серверу
+type Pusher interface {
```
Интерфейс называется `Pusher`, в то время как единственный метод в нем — `Send(...)`. Мне кажется, это немного неконсистентно. К тому же, в Go принято использовать для интерфейсов с одним методом наименование в формате `НазваниеМетода + er`.

vokinneberg reviewed (approved):
Поздравляю с успешным завершением четвертого спринта! 🎉 В этом спринте мы коснулись, наверное, самых сложных тем: многопоточности и криптографии. Ты реализовал доработки в соответствии с постановкой задачи. Отличная работа! 🚀 Желаю удачи на выпускном проекте! 🍀

**Сильные стороны твоего PR:**

1. Код функционален и корректен! 🚀
2. Твой код элегантный и лаконичный, его приятно и легко читать. 💪
3. Ты консистентно встроил новые функции в структуру существующего приложения. 👍
4. Замечаний практичесчки нет 🙌

Fairness2 committed:
iter15.4 | Переименован интерфейс Pusher в Sender.

Интерфейс Pusher и связанные с ним структуры и методы были переименованы в Sender для более точного описания их функциональности. Это изменение затронуло файлы mock, тесты и клиентский код.

Измененные файлы:
- cmd/agent/collector/sender/sender.go
- cmd/agent/collector/sender/mock/sender.go
- cmd/agent/collector/sender/sender_test.go
